mypy/semanal.py

```diff
@@ -690,8 +696,10 @@ def analyze_class_body(self, defn: ClassDef) -> Iterator[bool]:
                     if prohibited in named_tuple_info.names:
                         if nt_names.get(prohibited) is named_tuple_info.names[prohibited]:
                             continue
+                        ctx = named_tuple_info.names[prohibited].node
+                        assert ctx is not None
```

Comment:
> What if the definition is a bound type variable where ``node`` is ``None``, I think?

Currently generic named tuples are not supported, so I thought this should never happen.
I would keep this assert and then if it ever fails it will be clear what happens there.